src/zenml/post_execution/pipeline.py

```diff
@@ -72,10 +63,10 @@ def runs(self) -> List["PipelineRunView"]:
         """
         # Do not cache runs as new runs might appear during this objects
         # lifecycle
-        runs = list(self._metadata_store.get_pipeline_runs(self).values())
+        runs = list(Repository().zen_store.get_pipeline_runs(self).values())
```

Comment:
A quick explanation on why this contained the metadata store as a private attribute before:
```python
pipeline = Repository().get_pipeline(...)
# switch to a different profile
GlobalConfig().activate_profile("other_profile")
# the following call now fails or results in unexpected behaviour as it gets
# runs from a different profile and the associated zen store
# that's the issue with having this huge repo singleton that we query everywhere,
# changing the active state in a python process messes with a lot of things
pipeline.get_runs()
```

I'm not sure if this is a concern for now, but it could certainly bite us at some point. (Same for all the other post-execution classes)

Reply:
Yes, this will be an issue. However, I don't think switching profiles in the middle of your workflows should be supported anyways. I'd propose to check in each post execution class call that the active profile is the same as the profile that the class was queried from and raise an error otherwise. Any opinions?

Reply:
Seems like almost as much (if not more) work than actually supporting this feature?
However I agree that probably switching profiles should not be something that is happening while running code, however we won't be able to prevent users from doing that if they want. Anyway I'll leave this up to you, nothing critical imo.